validator/sawtooth_validator/networking/dispatch.py

```diff
@@ -147,14 +162,18 @@ def _process(self, message_id):
             handler_manager = next(collection)
             future = handler_manager.execute(connection_id, message.content)
 
-            def do_next(future):
+            timer_tag = type(handler_manager._handler).__name__
+            timer_ctx = self._get_dispatch_timer(timer_tag).time()
+
+            def do_next(timer_ctx, future):
+                timer_ctx.stop()
                 try:
                     self._determine_next(message_id, future)
                 except Exception:  # pylint: disable=broad-except
                     LOGGER.exception(
                         "Unhandled exception while determining next")
 
-            future.add_done_callback(do_next)
+            future.add_done_callback(functools.partial(do_next, timer_ctx))
```

Comment:
Heading this off before the comment:

https://docs.python.org/3/library/asyncio-eventloop.html#asyncio-pass-keywords